code/src/java/pcgen/cdom/primitive/CompoundAndPrimitive.java

```diff
@@ -58,7 +58,7 @@ public CompoundAndPrimitive(Collection<PrimitiveCollection<T>> pcfCollection)
 				Logging.log(Level.WARNING, "Found duplicate item in "
 						+ pcfCollection);
 			}
-			primCollection.add(PrimitiveCollection.INVALID);
+			primCollection.add(PrimitiveCollection.FIXED.<T>invalid());
```

Comment:
Yes, this is valid Java.  It's how one eliminates the generics warnings in this case.
As to need to be clever or not??  Just trying to reduce the warnings in our code, since generics can help catch problems too